runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+	"ForceExpression":       func() Expression { return &ForceExpression{} },
+	"FunctionExpression":    func() Expression { return &FunctionExpression{} },
+	"IdentifierExpression":  func() Expression { return &IdentifierExpression{} },
+	"IndexExpression":       func() Expression { return &IndexExpression{} },
+	"IntegerExpression":     func() Expression { return &IntegerExpression{} },
+	"InvocationExpression":  func() Expression { return &InvocationExpression{} },
+	"MemberExpression":      func() Expression { return &MemberExpression{} },
+	"NilExpression":         func() Expression { return &NilExpression{} },
+	"PathExpression":        func() Expression { return &PathExpression{} },
+	"ReferenceExpression":   func() Expression { return &ReferenceExpression{} },
+	"StringExpression":      func() Expression { return &StringExpression{} },
+	"UnaryExpression":       func() Expression { return &UnaryExpression{} },
+}
+
+var locationKindMap = map[string]func() common.Location{
+	"AddressLocation":     func() common.Location { return &common.AddressLocation{} },
+	"REPLLocation":        func() common.Location { return &common.REPLLocation{} },
+	"ScriptLocation":      func() common.Location { return &common.ScriptLocation{} },
+	"TransactionLocation": func() common.Location { return &common.TransactionLocation{} },
+	"StringLocation":      func() common.Location { return nil },
+	"FlowLocation":        func() common.Location { return nil },
+	"IdentifierLocation":  func() common.Location { return nil },
+}
+
+var operationMap = map[string]Operation{
+	"OperationUnknown":           OperationUnknown,
+	"OperationOr":                OperationOr,
+	"OperationAnd":               OperationAnd,
+	"OperationEqual":             OperationEqual,
+	"OperationNotEqual":          OperationNotEqual,
+	"OperationLess":              OperationLess,
+	"OperationGreater":           OperationGreater,
+	"OperationLessEqual":         OperationLessEqual,
+	"OperationGreaterEqual":      OperationGreaterEqual,
+	"OperationPlus":              OperationPlus,
+	"OperationMinus":             OperationMinus,
+	"OperationMul":               OperationMul,
+	"OperationDiv":               OperationDiv,
+	"OperationMod":               OperationMod,
+	"OperationNegate":            OperationNegate,
+	"OperationNilCoalesce":       OperationNilCoalesce,
+	"OperationMove":              OperationMove,
+	"OperationCast":              OperationCast,
+	"OperationFailableCast":      OperationFailableCast,
+	"OperationForceCast":         OperationForceCast,
+	"OperationBitwiseOr":         OperationBitwiseOr,
+	"OperationBitwiseXor":        OperationBitwiseXor,
+	"OperationBitwiseAnd":        OperationBitwiseAnd,
+	"OperationBitwiseLeftShift":  OperationBitwiseLeftShift,
+	"OperationBitwiseRightShift": OperationBitwiseRightShift,
+}
+
+var statementKindMap = map[string]func() Statement{
+	"AssignmentStatement":        func() Statement { return &AssignmentStatement{} },
```

Comment:
What is the reason for the value being a function here? Can the value type of the map just be Expression?